front/app/modules/commercial/insights/admin/containers/Insights/Details/Network/index.tsx

```diff
@@ -77,18 +90,41 @@ const StyledMessage = styled.h4`
 `;
 
 const Network = ({
-  params: { viewId },
   intl: { formatMessage, formatDate },
-  location: { query, pathname },
-}: WithRouterProps & InjectedIntlProps) => {
+}: InjectedIntlProps) => {
   const [initialRender, setInitialRender] = useState(true);
   const [height, setHeight] = useState(0);
   const [width, setWidth] = useState(0);
   const [zoomLevel, setZoomLevel] = useState(0);
+  const [hoverNode, setHoverNode] = useState<Node | undefined>();
+  const [pointerPosition, setPointerPosition] = useState([0, 0]);
+  const [graphInitialized, setGraphInitialized] = useState<boolean>(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { viewId } = useParams() as { viewId: string };
 
   const networkRef = useRef<ForceGraphMethods>();
   const { loading, network } = useNetwork(viewId);
   const view = useInsightsView(viewId);
+  const tooltipRef = document.getElementsByClassName('graph-tooltip')[0];
+
+  const setPointerEvent = (e) => setPointerPosition([e.offsetX, e.offsetY]);
```

Comment:
I think it's better to use `useRef` for this instead of state. This function is going to be called a LOT (every time the mouse moves a pixel) and the `setPointerPosition` will trigger a rerender every time. Try putting a `console.log` somewhere in the component with the current method, and then try replacing the state by a ref and you will see much let `console.log`s!

Reply:
`useRef` its actually very slow for updating the mouse coordinates and the hidding label doesn't change correctly. So while its true that its creating a lot of rerenders its the best way to change a node label quickly when hovering the close icon.